network/gossip/libp2p/libp2pConnector.go

```diff
@@ -0,0 +1,109 @@
+package libp2p
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	discovery "github.com/libp2p/go-libp2p-discovery"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// libp2pConnector is a libp2p based Connector implementation to connect and disconnect from peers
+type libp2pConnector struct {
+	backoffConnector *discovery.BackoffConnector
```

Comment:
https://github.com/libp2p/go-libp2p-discovery/blob/master/backoffconnector.go
"// BackoffConnector is a utility to connect to peers, but only if we have not recently tried connecting to them already"